openassessment/workflow/models.py

```diff
@@ -316,17 +322,23 @@ def update_from_assessments(self, assessment_requirements):
 
         new_staff_score = self.get_score(assessment_requirements, {'staff': step_for_name.get('staff', None)})
         if new_staff_score:
-            old_score = self.score
+            old_score = sub_api.get_latest_score_for_submission(self.submission_uuid)
             if not old_score or old_score['points_earned'] != new_staff_score['points_earned']:
+                # Set the staff score using submissions api, and log that fact
                 self.set_staff_score(new_staff_score)
                 self.save()
                 logger.info((
                     u"Workflow for submission UUID {uuid} has updated score using staff assessment."
                 ).format(uuid=self.submission_uuid))
-                staff_step = step_for_name.get('staff')
-                staff_step.assessment_completed_at=now()
-                staff_step.save()
-                self.status = self.STATUS.done
+
+                # Update the assessment and submitter_completed_at fields for all steps
+                # All steps are considered "assessment complete", as the staff score will override all
+                # Steps in BLOCKING_STEPS may still require the submitter to fulfill their obligations
+                for step in steps:
```

Comment:
Honestly, this part is also confusing me, but talking over it in person tomorrow with the new version will likely clear it up for me.